compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/Accessibility.uikit.kt

```diff
@@ -929,13 +991,17 @@ internal class AccessibilityMediator(
     init {
         getAccessibilitySyncOptions().debugLoggerIfEnabled?.log("AccessibilityMediator for ${view} created")
 
-        val updateIntervalMillis = 50L
-        // TODO: this approach was copied from desktop implementation, obviously it has a [updateIntervalMillis] lag
-        //  between the actual change in the semantics tree and the change in the accessibility tree.
-        //  should we use some other approach?
         coroutineScope.launch {
             while (isAlive) {
-                var result: NodesSyncResult
+                var strategy = SemanticsTreeSyncStrategy.from(
+                    invalidationChannel.receive()
+                )
+
+                while (true) {
+                    // Consumes all the invalidations to select a proper strategy
+                    val invalidation = invalidationChannel.tryReceive().getOrNull() ?: break
+                    strategy = strategy.reduce(invalidation)
+                }
```

Comment:
Extract this into a helper function (channel ext)

Reply:
I'd like to keep this code inline so the entire `suspend`->`consumeAll` sequence is visible to a reader

Reply:
How is "suspend" related here?

Reply:
The `receive` call is two lines above, it's suspending and is a delimiter for a synchronous inner buffer consumption.
My intent was to make this entire logic visible to a reader.

Reply:
Reverted to pre-refactor